### ISSUE TITLE
installing `pytest` doesn't add `requests`, `selenium`, or `testpath`

### ISSUE THREAD

**Issue by acalvino4** [pdm-project/pdm#1444]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

## Steps to reproduce

- pdm add pytest
- pdm run pytest

## Actual behavior

It seems that installing pytest does not install its dependencies

```
platform darwin -- Python 3.10.7, pytest-7.1.3, pluggy-1.0.0
rootdir: /Users/acalvino4/python
plugins: anyio-3.6.1
collected 0 items / 1 error                                  

=========================== ERRORS ===========================
_______________ ERROR collecting test session ________________
../.pyenv/versions/3.10.7/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
__pypackages__/3.10/lib/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
__pypackages__/3.10/lib/nbclassic/tests/selenium/conftest.py:6: in <module>
    import requests
E   ModuleNotFoundError: No module named 'requests'
================== short test summary info ===================
ERROR  - ModuleNotFoundError: No module named 'requests'
!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!
====================== 1 error in 2.19s ======================
```

## Expected behavior

pytest to run my tests

## Environment Information

```bash
pdm info && pdm info --env

PDM version:
  2.1.5
Python Interpreter:
  /Users/acalvino4/.pyenv/versions/3.10.7/bin/python (3.10)
Project Root:
  /Users/acalvino4/python
Project Packages:
  /Users/acalvino4/python/__pypackages__/3.10
{
  "implementation_name": "cpython",
  "implementation_version": "3.10.7",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "21.6.0",
  "platform_system": "Darwin",
  "platform_version": "Darwin Kernel Version 21.6.0: Mon Aug 
22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64",
  "python_full_version": "3.10.7",
  "platform_python_implementation": "CPython",
  "python_version": "3.10",
  "sys_platform": "darwin"
}
```

## Other

```bash
$ echo $PYTHONPATH

/usr/local/Cellar/pdm/2.1.5/libexec/lib/python3.10/site-packages/pdm/pep582

$ pdm config
...
Home configuration (/Users/acalvino4/Library/Preferences/pdm/config.toml):
install.cache = True
strategy.save = compatible

Project configuration (/Users/acalvino4/python/.pdm.toml):
python.path = /Users/acalvino4/.pyenv/shims/python
```

`pyproject.toml`
```toml
[project]
...
dependencies = [
]
requires-python = ">=3.10"

...

[tool.pdm]
[tool.pdm.dev-dependencies]
dev = [
    "black~=22.10.0",
    "flake8~=5.0.4",
    "flake8-bandit~=4.1.1",
    "flake8-docstrings~=1.6.0",
    "flake8-isort~=5.0.0",
    "flake8-pyproject~=1.1.0.post0",
    "mypy~=0.982",
    "notebook~=6.5.1",
    "pytest~=7.1.3",
    "types-setuptools~=65.5.0.1",
]
```

**Comment by pawamoy** [comment]
Neither requests, testpath or selenium are pytest dependencies. Is that why you added those explicitly in pyproject.toml? Then did you resolve and install your dependencies with `pdm install`, or `pdm lock` then `pdm install`?

**Comment by acalvino4** [comment]
Whoops, that was a modification I made to try and get it to work, and it did. I received the error when I didn't have those in `dependencies`, and I've updated original post to reflect that.

And I did resolve dependencies with `pdm update` and `pdm install`.

Anyway, I'm not having the issue anymore (I don't get the error message with or without `requests` installed). I'm not sure what I changed that fixed it, but if I can replicate with more clear steps I'll reopen.
